library/firewall_lib.py

```diff
@@ -178,7 +178,39 @@
 except ImportError:
     HAS_FIREWALLD = False
 
-
+def handle_interface(zone,item,fw_zone,fw_settings,fw,fw_offline):
+    if fw_offline:
+        iface_zone_objs = []
+        for zone in fw.config.get_zones():
+            old_zone_obj = fw.config.get_zone(zone)
+            if item in old_zone_obj.interfaces:
+                iface_zone_objs.append(old_zone_ob)
```

Comment:
old_zone_obj?

(Not really reviewing this, Rich's comments just landed in my inbox and this caught my eye.)

Also, I'm surprised this seems to have passed the checks.
Don't you run some kind of static analysis? pylint?